### ISSUE TITLE
canvas flashes with default position/zoom before moving

### ISSUE THREAD

**Issue by vbud** [aarondail/react-zoomable-ui#49]
I'm using `onCreate` to give the canvas a starting position/zoom that is different than the default position/zoom. According to the [documentation](https://github.com/aarondail/react-zoomable-ui/blob/master/docs/api/interfaces/spaceprops.md#optional-oncreate), `onCreate` is supposed to give me an opportunity to move the viewport around before the inner div is rendered. However, am seeing the inner div render before the viewport updates, resulting in a flash of incorrectly positioning children before they snap to the correct location.

My code looks like this, nothing fancy: 
```
onCreate={(viewPort) => {
  const { left, top, zoom } = canvasPosition;
  viewPort.camera.updateTopLeft(left, top, zoom);
}}
```

Any idea why it isn't worked as expected?

**Comment by aarondail** [comment]
Thanks for filing the issue! Crap yeah... not sure how I missed this but I see it. There is some code in the `Space` class that will delay rendering the children you give it until after the `onCreate` is run, and I think that is working, so I think the problem is probably how updates to the camera (`viewPort.camrea.updateTopLeft(...)`) actually get executed. Instead of them happening synchronously like you might expect they get "queued" (so to speak) via `requestAnimationFrame`. Probably the callback to `requestAnimationFrame` is then executing (at least some of the time) after the React rendering, thus the flash. (IIRC I did things that way to make the animated and non-animated camera update code more similar.)

Ill try to dig into it more this week and maybe get a fix out.

If you are blocked I think a somewhat hacky fix you could do (assuming Im on the right path here) is in your component to add a boolean to your component state (`const initialFlashHackyFix = useState(false)`), render nothing inside the `Space`, if that is false, then in your onCreate set pass a callback to `setTimer` or `requestAnimationFrame` to set it to true. Something like that.

**Comment by vbud** [comment]
That makes sense. I pulled this repo down locally too and yup, seems to be the animation frame issue.

Building on the workaround you suggested, I noticed that [`onUpdated` is called after the camera is updated](https://github.com/aarondail/react-zoomable-ui/blob/master/src/ViewPortCamera.ts#L412), so I set a flag to `true` once I see that first `onUpdated` call. Seems to work.

```jsx
<Space
	onCreate={(viewport) => {
		const { left, top, zoom } = canvasPosition;
		viewport.camera.updateTopLeft(left, top, zoom);
	}}
	onUpdated={(vp) => {
		if (!isViewportReady) {
			setIsViewportReady(true);
		}
	}}
>
	{isViewportReady && children}
</Space>
```

Let me know if you end up getting a fix out! In the meantime, this solution is working well for me.

**Comment by aarondail** [comment]
Cool, glad that worked. I had some time to work on this so I released a fix for the problem just now in `0.11.0`. This should eliminate the need for your workaround.

Lmk if you have a chance to check it out and if it resolves the issue for you or not.

The fix was just to make non-animated camera updates happen synchronously (no `requestAnimationFrame`). Its not impossible that will have some other consequence that I can't forsee... but so far I haven't found anything while testing. Lmk if you hit anything.

**Comment by vbud** [comment]
Working great, thanks so much for the fix!

I'll let you if I notice any issues, but I've tried everything I can think of and haven't noticed anything weird.
